init.c

```diff
@@ -57,5 +58,19 @@ int main(int argc, char *argv[], char *envp[])
 			}
 			_exit(exitcode);
 		}
+		if (WIFSTOPPED(status)) {
+			/*
+			 * Empirically, if a traced process's parent exits, the
+			 * init process inherits the tracing of that process.
+			 * If we notice an inherited child has stopped without
+			 * explicitly asking for that notification, detach from
+			 * it, forwarding WSTOPSIG.
+			 */
+			int rc = ptrace(PTRACE_DETACH, pid, 0, WSTOPSIG(status));
```

Comment:
Isn't the data arg of ptrace ignored with PTRACE_DETACH? what's the purpose of passing in WSTOPSIG(status)?

Reply:
No - "addr" is ignored. "data" is the signal you want to send to the target.
```
       PTRACE_DETACH
              Restart the stopped tracee as for PTRACE_CONT, but first detach from it.  Under Linux, a tracee can be detached in this way regardless  of
              which method was used to initiate tracing.  (addr is ignored.)
```
And:
```
       PTRACE_CONT
              Restart the stopped tracee process.  If data is nonzero, it is interpreted as the number of a signal to be delivered to the tracee; other‐
              wise, no signal is delivered.  Thus, for example, the tracer can control whether a signal sent to the tracee is delivered or  not.   (addr
              is ignored.)
```